### ISSUE TITLE
Fail to open exported file in Excel, if web server is hosted on Linux

### ISSUE THREAD

**Issue by majoshi1** [ag-grid/ag-grid#2004]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
The exported file from demo does not open in Excel, if web server is hosted on Linux.
Works fine when application is hosted on Windows.

**Expected behavior**
The file exported as per the demo should open, for all hosting environments.

**Minimal reproduction of the problem with instructions**
Host following demo in a web server on Linux.
https://www.ag-grid.com/example-runner/vanilla.php?section=javascript-grid-excel&example=custom-xlsx&enterprise=1
Optional: Change to use getDataAsCsv (for free version) instead of getDataAsExcel.
``
var content = gridOptions.api.getDataAsCsv(params);
``

**What is the motivation / use case for changing the behavior?**
Ability to get XLSX file generated from the application, hosted on Linux.

**Workaround (if any)**
Use xml, instead of xlsx format. 
Modify the demo, with following changes:
-Change xlsx to xml
``
var xlsxContent = XLSX.write(workbook, { bookType:'xml', . . .
``
-Change xlsx to xls
``
var fileName = fileNamePresent ? params.fileName : 'noWarning.xls';
``
-Change type
``
var blobObject = b64toBlob(content, 'application/octet-stream');
``

**Please tell us about your environment:**
OS: Linux

* **ag-Grid version:**
Any

* **Browser:**
Any

* **Language:** 
all

**Comment by petyosi** [comment]
Hi, 

the ag-grid website itself is hosted on a linux OS. The grid component is a client-side only library, so the server OS itself is unlikely to affect its behavior. The problem might be related to a web server configuration, though. 

To get further assistance with the matter, I may suggest that you post your question in [our forum](https://ag-grid.com/forum/forumdisplay.php?fid=5). We will look into it further.

**Comment by majoshi1** [comment]
Found the cause of this issue. 
The application uses [(Jawr)](https://j-a-w-r.github.io/) for bundling of resources (including JavaScript files). 
The issue occurs, on any OS, when all the JS files are bundled into single file (UTF-8). 
It works fine, if bundling is disabled.
``
jawr.debug.on=true
``
But, that is not recommended in production.
So, will have to go with the workaround for now.

It will be interesting to see the behavior with other file bundling tools like angular-cli.

**Comment by majoshi1** [comment]
Found the root cause of this issue.
I picked minified version of SheetJS library from the demo.
The Jawr tool also minifies the file, before bundling.
That must have messed up the file.

Steps to fix the issue:
-Removed the minified SheetJS library file (xlsx.core.min-0.10.3.js).
-Added non-minified files: xlsx.js and jszip.js from [SheetJS](https://github.com/SheetJS/js-xlsx/tree/v0.10.8/dist) github.
Now, it works for XLSX format also.

Sorry for the confusion.

PS: The XLSX library was formatting numbers as dates.
So, added workbook option: `` raw: true `` as per [this](https://github.com/SheetJS/js-xlsx/issues/719) post.
Also, it needed to upgrade from 0.10.3 to 0.10.8, as per the same post.

And if you are using getDataAsCsv, remove double quotes.

```
var content = gridOptions.api.getDataAsCsv(params);
content = content.replace(/\"/gm, ""); // Remove double quotes
var workbook = XLSX.read(content, {type: 'binary', raw: true}); // Raw
var xlsxContent = XLSX.write(workbook, {bookType: 'xlsx', type: 'base64'});

```
